### PR TITLE
fix for issue #430 where " or ' exist as valid data adjacent right to…

### PR DESCRIPTION
**Pull request by lunatixcoder**
In some cases data contains leading quotes formed by " or ' and are right adjacent to the delimiter. 

Example:

060003T,RACEAIR DESIGNS INC ,'LIL BITTS ,4,7 ,1,2,01,001,CLASS 1,0000,

05618AR,MILLER ,"A" ,4,1 ,1,1,01,001,CLASS 1,0085,

05618AT,TRAVEL AIR ,"R" ,4,1 ,1,1,01,001,CLASS 1,0085,

In these cases we want to treat all data between commas as data including those characters.  So we need a way to tell papaparse, with positive assertion, to ignore quotes.  However papaparse takes all false statements and defaults the quoteChar field to " which creates issues parsing the data when the above data example is present.

With this change, if quoteChar is not specified in the papaparse config, it evaluates to undefined which is distinctly checkable so we can keep behavior as it.  This then allows you to not specify quoteConfig and it remains at defaulting to " otherwise if you positively assert quoteChar: "" then it tells papaparse to ignore quotes by saying there are none.

### CODE REVIEW

File: papaparse.js [hunk 1]
```diff
@@ -269,7 +269,25 @@
 		var _newline = '\r\n';
 
 		/** quote character */
-		var _quoteChar = '"';
+// lunatixcoer: Changed logic so you can ignore quote characters altogether.
```
Comment (pokoli) [inline papaparse.js]: I will prefer to have a simpler comment. Like: "Use default value only if it's not set".

File: papaparse.js [hunk 2]
```diff
@@ -269,7 +269,25 @@
...
+//				we can check for undefined and default it to """, 
+//              otherwise we should take what literally it is set to or if set to "" ignore quotes altogether.
+//
+// Lunatixcoder: removed...
+//
+//      var quoteChar = config.quoteChar || '"';
+//
+//      and replaced with 	
+
+		if ( config.quoteChar == undefined ) {
```
Comment (pokoli) [inline papaparse.js]: Better to use ===

File: papaparse.js [hunk 3]
```diff
@@ -269,7 +269,25 @@
...
+			var quoteChar = '"'; // sets default value to " so it should not effect existing code used by other programs
```
Comment (pokoli) [inline papaparse.js]: There is no need to comment every line.

File: papaparse.js [hunk 4]
```diff
@@ -269,7 +269,25 @@
...
+		} else {
+			var quoteChar = config.quoteChar; // that way if you set quoteChar in your config to a positive assertion of quoteChar: "", it will ignore quotes between delimiters
```
Comment (pokoli) [inline papaparse.js]: Idem about comments

**Comment by pokoli** [comment]
I can not merge it untill tests are fixed. 

BTW, it will be great to have a unit test for this case.

**Comment by lunatixcoder** [comment]
Ok... I think I made the changes you requested.  Im new to pull requests so Im hoping Im doing this correctly.  Can you see the changes I made per your request to change comments?

File: papaparse.js (hunk 2 above)
Comment (pokoli) [reply papaparse.js]: This comment is still missing

**Comment by pokoli** [comment]
THere is one change missing. 

Also have a look at tests that are failing:

https://travis-ci.org/mholt/PapaParse/builds/298513060


You can run test localy with: 

`npm test`

Just make sure you have node installed.

File: papaparse.js [hunk 5]
```diff
@@ -269,7 +269,13 @@
 		var _newline = '\r\n';
 
 		/** quote character */
-		var _quoteChar = '"';
+
```
Comment (pokoli) [inline papaparse.js]: Empty line can be removed.

**Comment by lunatixcoder** [comment]
Ok... so doing NPM test on my machine worked.  one thing I did was downgrade from 9.0 to 8.9 of Node because NPM said it wasnt compatible with 9.0. Did I do something wrong?

File: package.json [hunk 6]
```diff
@@ -56,5 +56,8 @@
 		"test-phantomjs": "node tests/test.js --phantomjs",
 		"test-node": "mocha tests/node-tests.js tests/test-cases.js",
 		"test": "npm run test-node && npm run test-phantomjs"
+	},
```
Comment (pokoli) [inline package.json]: This should not be included here.

**Comment by lunatixcoder** [comment]
I think the issue was that I DID remove those and it caused it to fail because I used 8.9.0  I reinstalled 9.0.0 and re committed and I think that will pass

**Comment by lunatixcoder** [comment]
Ughhh it didnt. let me try what you suggested

**Comment by lunatixcoder** [comment]
YAY!

**Comment by pokoli** [comment]
Great tests are green. Could you please remove the non relevant files and include only the changes to papaparse.js file?

Then we will be done.

**Comment by lunatixcoder** [comment]
Im not familiar enough with GitHub to understand what you wanted me to do I think.  Or was this what you wanted? Again, this is my first time doing this.

**Comment by pokoli** [comment]
Ok, don't worry. Just be pattient. 

I don't mean you have to remove all the files from the project. But include on the PR only the changes relevant to the PR. 

Don't worry, I'm going to merge it manually. 

Thanks for your time.

**Comment by lunatixcoder** [comment]
Thanks for being so kind and patient with me. I really want to learn this and I know you are not here to teach me.  I think I know what I did wrong if you want me to try again.

**Comment by pokoli** [comment]
I merged it manually on d9c78a04ba046ff097bd77730c9af7ac8686a980

Thanks for your contribution. 

Indeed, I will accept a pull request to explain the change on the docs which are on the gh-pages branch. So you can retry the process :) But just to make sure to include only the minimum changes required to document the behaviour.

**Comment by lunatixcoder** [comment]
I cant figure out how to create a PR with GitHub on the documentation branch.  I cloned it and went to open a pull request and it said I did not have access to publish to that repository.

David

**Comment by pokoli** [comment]
You have to push to your fork and then createa pull request to this repository.

**Comment by lunatixcoder** [comment]
That is the problem.  It wont let me push to my fork.

This is a screen snapshot of what I get when I click on Push origin.

**Pull request closed without merging**